models/db/context.go

```diff
@@ -155,3 +127,28 @@ func Insert(ctx context.Context, beans ...interface{}) error {
 	_, err := GetEngine(ctx).Insert(beans...)
 	return err
 }
+
+// Exec executes a sql with args
+func Exec(ctx context.Context, sqlAndArgs ...interface{}) (sql.Result, error) {
+	return GetEngine(ctx).Exec(sqlAndArgs...)
+}
+
+// GetByExample filled empty fields of the bean according non-empty fields to query in database.
+func GetByExample(ctx context.Context, bean interface{}) (bool, error) {
```

Comment:
We already have concepts like `model`, `bean`, here we will get a new concept `example`.

Can these concepts be unified?

Reply:
Perhaps a package called `crud`?

```go
package crud

// Get bean matching non-empty fields.
func Get(ctx context.Context, bean interface{}) (bool, error) {
  return db.GetEngine(ctx).Get(bean)
}

// Update bean with id matching non-empty fields. Passing id = 0 will update all records
func Update(ctx context.Context, bean interface{}, id int64) (int64, error) {
  if id != 0 {
     return db.GetEngine(ctx).ID(id).Update(bean)
  }
  return db.GetEngine(ctx).Update(bean)
}

// Delete all beans with matching non-empty fields
func Delete(ctx context.Context, bean interface{}) (int64, error) {
  return db.GetEngine(ctx).Delete(bean)
}

// Count all beans with matching non-empty fields
func Count(ctx context.Context, bean interface{}) (int64, error) {
  return db.GetEngine(ctx).Count(bean)
}

// TableName returns the table name for a provided bean
func TableName(ctx context.Context, bean interface{}) string {
  return db.GetEngine(ctx).TableName(bean)
}

```

Reply:
I think put them in db is enough. And about the name, `Get` or `GetByExample`, which one do you like?